smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/events/MessageUnmarshallableGenerator.kt

```diff
@@ -0,0 +1,210 @@
+package software.amazon.smithy.swift.codegen.events
+
+import software.amazon.smithy.codegen.core.CodegenException
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.ShapeType
+import software.amazon.smithy.model.shapes.UnionShape
+import software.amazon.smithy.model.traits.EventHeaderTrait
+import software.amazon.smithy.model.traits.EventPayloadTrait
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.SwiftDependency
+import software.amazon.smithy.swift.codegen.SwiftTypes
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.HTTPProtocolCustomizable
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.serde.readwrite.ReadingClosureUtils
+import software.amazon.smithy.swift.codegen.integration.serde.struct.readerSymbol
+import software.amazon.smithy.swift.codegen.model.eventStreamErrors
+import software.amazon.smithy.swift.codegen.model.eventStreamEvents
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.model.hasTrait
+
+class MessageUnmarshallableGenerator(
+    val ctx: ProtocolGenerator.GenerationContext,
+    val customizations: HTTPProtocolCustomizable,
```

Comment:
This class now takes customizations as a parameter, so that the unknown error type can be passed in by the calling protocol.